packages/storybook/src/schematics/migrate-defaults-5-to-6/migrate-defaults-5-to-6.ts

```diff
@@ -0,0 +1,167 @@
+import {
+  chain,
+  move,
+  Rule,
+  SchematicContext,
+  template,
+  Tree,
+  url,
+} from '@angular-devkit/schematics';
+import { offsetFromRoot, readWorkspaceJson } from '@nrwl/workspace';
+import {
+  applyWithSkipExisting,
+  getProjectConfig,
+} from '@nrwl/workspace/src/utils/ast-utils';
+import { projectDir } from '@nrwl/workspace/src/utils/project-type';
+import { StorybookMigrateDefault5to6Schema } from './schema';
+
+export default function (schema: StorybookMigrateDefault5to6Schema): Rule {
+  /**
+   * The "all" flag - or the absense of "name"
+   * should indicate that all Storybook instances everywhere in the
+   * project should be migrated.
+   *
+   * Not running a migration for "all" does NOT make
+   * sense, since everything links back to the root .storybook
+   * directory, which will get migrated.
+   * However, someone may want to do it step by step. But all should be migrated
+   * eventually.
+   */
+  return chain([
+    schema.all === false
+      ? migrateStorybookInstance(
+          schema.keepOld,
+          schema.name,
+          schema.uiFramework
+        )
+      : migrateAllStorybookInstances(schema.keepOld),
+    migrateRootLevelStorybookInstance(),
+    upgradeStorybookPackagesInPackageJson(),
+  ]);
+}
+
+export function migrateAllStorybookInstances(keepOld: boolean): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    context.logger.debug('adding .storybook folder to lib');
+    const workspaceJson = readWorkspaceJson();
+    const projects: string[] = Object.keys(workspaceJson.projects);
+    const projectsThatHaveStorybookConfiguration: {
+      name: string;
+      uiFramework: string;
+      configFolder: string;
+    }[] = projects.map((project: string) => {
+      if (
+        workspaceJson.projects[project].targets &&
+        workspaceJson.projects[project].targets.storybook
+      )
+        return {
+          name: project,
+          uiFramework:
+            workspaceJson.projects[project].targets.storybook.options
+              .uiFramework,
+          configFolder:
+            workspaceJson.projects[project].targets.storybook.options.config
```

Comment:
when I tried this, all usages of `.targets` failed because I was using `architect`.

Should we cater for both?